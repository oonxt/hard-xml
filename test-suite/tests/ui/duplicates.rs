use hard_xml::XmlRead;
use std::borrow::Cow;

#[derive(Debug, XmlRead)]
#[xml(tag = "Attributes")]
struct Attributes<'t> {
    #[xml(attr = "field1", attr = "field2")]
    field: Cow<'t, str>,
}

#[derive(Debug, XmlRead)]
#[xml(tag = "Cdata")]
struct Cdata<'t> {
    #[xml(text, cdata, cdata)]
    field: Cow<'t, str>,
}

#[derive(Debug, XmlRead)]
#[xml(tag = "Child")]
struct FlattenText<'t> {
    #[xml(flatten_text = "field1", flatten_text = "field2")]
    field: Cow<'t, str>,
}

#[derive(Debug, XmlRead)]
#[xml(tag = "Text")]
struct Text<'t> {
    #[xml(text, text)]
    field: Cow<'t, str>,
}

fn main() {}
