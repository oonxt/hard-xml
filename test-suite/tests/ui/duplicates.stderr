error: duplicate `attr` attribute
 --> tests/ui/duplicates.rs:7:28
  |
7 |     #[xml(attr = "field1", attr = "field2")]
  |                            ^^^^

error: Duplicate `cdata` attribute.
  --> tests/ui/duplicates.rs:14:24
   |
14 |     #[xml(text, cdata, cdata)]
   |                        ^^^^^

error: Duplicate `flatten_text` attribute.
  --> tests/ui/duplicates.rs:21:36
   |
21 |     #[xml(flatten_text = "field1", flatten_text = "field2")]
   |                                    ^^^^^^^^^^^^

error: Duplicate `text` attribute.
  --> tests/ui/duplicates.rs:28:17
   |
28 |     #[xml(text, text)]
   |                 ^^^^
